File: web-dev/cs50/intro-comp-science/Problem-sets/pset1/mario_less/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    //Prompting the user for an input
    int height;
    do
    {
        height = get_int("Height for the pyramid: ");
    }
    // This checks if the given input from the user fulfills our condition (i.e. between 1      & 8 inclusive).
    while (height < 1 || height > 8);

    // Iteration for columns up until the given input from the user has been reach.
    for (int i = 0; i < height; i++)
    {
        // This will print empty spaces helping to realign our blocks to the right.
        for (int k = (height - i) - 1; k >= 1; k--)
        {
            printf(" ");
        }
        // This prints the HASH symbol up until re reach the input from the user.
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }
        // This will print a new line after each iteration prints
        printf("\n");
    }

}

File: web-dev/cs50-2021/C-Program/week6/mario/less/mario.py
import cs50


def main():
    while True:
        # Get an input from the user (defined as height for the pyramid)
        height = cs50.get_int("Height: ")

        # Set a condition that user's input must fulfill
        if height > 0 and height <= 8:
            break
    # Calling our function that will draw the pyramid
    mario(height)

# Defining a function that draws the pyramid ()


def mario(n):
    for i in range(n):
        print(' ' * (n - i - 1) + '#' * (i + 1))


main()
